pkg/install/install.go

```diff
@@ -396,8 +396,8 @@ func (i *Installer) deployARMTemplate(ctx context.Context, rg string, tName stri
 		return err == nil, err
 	}, timeoutCtx.Done())
 
-	if isQuota, errMsg := hasResourceQuotaExceededError(err); isQuota {
-		err = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeQuotaExceeded, errMsg, "")
+	if err != nil {
+		err = api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", err.Error())
```

Comment:
`az` and many API clients will automatically retry if they get a 500.  I think you should return StatusBadRequest and a new CloudErrorCode, something like CloudErrorCodeDeploymentFailed.  That error should be structured such that a caller can pick it apart and see what happened.  That's the tricky bit of this, i.e. don't collapse it to a single string